src/sentry/replays/endpoints/organization_issue_replay_count.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from typing import Any
+
+from rest_framework import status
+from rest_framework.response import Response
+from snuba_sdk import Column, Condition, Op, Request
+
+from sentry import features
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import NoProjects
+from sentry.api.bases.organization_events import OrganizationEventsV2EndpointBase
+from sentry.models import Organization
+from sentry.replays.query import query_replays_dataset
+from sentry.search.events.builder import QueryBuilder
+from sentry.utils.snuba import Dataset, raw_snql_query
+
+
+@region_silo_endpoint
+class OrganizationIssueReplayCountEndpoint(OrganizationEventsV2EndpointBase):
+    """
+    Get all the replay ids associated with a set of issues in discover,
+    then verify that they exist in the replays dataset, and return the count.
+    """
+
+    private = True
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        if not features.has("organizations:session-replay", organization, actor=request.user):
+            return Response(status=404)
+
+        try:
+            params = self.get_snuba_params(request, organization, check_global_views=False)
+        except NoProjects:
+            return Response({})
+
+        try:
+            replay_id_to_issue_map = _query_discover_for_replayIds(request, params)
+        except ValueError as e:
+            return Response({"detail": str(e)}, status=status.HTTP_400_BAD_REQUEST)
+
+        replay_results = query_replays_dataset(
+            project_ids=params["project_id"],
+            start=params["start"],
+            end=params["end"],
+            sorting=[],
+            where=[
+                Condition(Column("replay_id"), Op.IN, list(replay_id_to_issue_map.keys())),
```

Comment:
Because this can potentially contain 2,500 replay-ids, have you done any performance testing with this?  Will this degrade query performance for other replay endpoints?

Reply:
I haven't done any performance testing, but I don't think it will be an issue based on my understanding of clickhouse query performance. -- can monitor upon rollout though. can also talk through my performance rationale if you'd like!